CHANGELOG.md

```diff
@@ -19,6 +19,15 @@
   will be removed in a future release. More details on the motivation for this change can be found
   in [ADR-0004](https://github.com/mozilla/uniffi-rs/blob/main/docs/adr/0004-only-threadsafe-interfaces.md).
 
+### What's New
+
+- It is now possible to use Object instances as fields in Records or Enums, to pass them as arguments,
+  and to return them from function and method calls. They should for the most part behave just like
+  a host language object, and their lifecycle is managed transparently using Rust's `Arc<T>` type.
+    - Cross-language reference cycles will not be garbage collected.
```

Comment:
The "cross language" part here makes it sound like some cycles will be collected - don't have a concrete suggestion here because we don't want to give the impression we will somehow break existing GC, but thought I'd mention it in case you have inspiration.